sdk/data/azcosmos/emulator_cosmos_item_test.go

```diff
@@ -503,6 +503,213 @@ func TestItemIdEncodingComputeGW(t *testing.T) {
+		t.Fatalf("Failed to create container: %v", err)
+	}
+
+	container, err := database.NewContainer("aContainer")
+	if err != nil {
+		t.Fatalf("Failed to get container: %v", err)
+	}
+
+	itemAlpha := map[string]interface{}{
+		"id":    "1",
+		"type":  "alpha",
+		"value": "0",
+	}
+
+	itemBeta := map[string]interface{}{
+		"id":    "1",
+		"type":  "beta",
+		"value": "0",
+	}
+
+	pkAlpha := NewPartitionKey().AppendString("1").AppendString("alpha")
+	pkBeta := NewPartitionKey().AppendString("1").AppendString("beta")
+
+	marshalledAlpha, err := json.Marshal(itemAlpha)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	marshalledBeta, err := json.Marshal(itemBeta)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	item0Res, err := container.CreateItem(context.TODO(), pkAlpha, marshalledAlpha, nil)
+	if err != nil {
+		t.Fatalf("Failed to create item: %v", err)
+	}
+
+	if item0Res.SessionToken == nil {
+		t.Fatalf("Session token is empty")
+	}
+
+	if len(item0Res.Value) != 0 {
+		t.Fatalf("Expected empty response, got %v", item0Res.Value)
+	}
+
+	item1Res, err := container.CreateItem(context.TODO(), pkBeta, marshalledBeta, nil)
+	if err != nil {
+		t.Fatalf("Failed to create item: %v", err)
+	}
+
+	if item1Res.SessionToken == nil {
+		t.Fatalf("Session token is empty")
+	}
+
+	if len(item1Res.Value) != 0 {
+		t.Fatalf("Expected empty response, got %v", item1Res.Value)
+	}
+
+	item0Res, err = container.ReadItem(context.TODO(), pkAlpha, "1", nil)
+	if err != nil {
+		t.Fatalf("Failed to read item: %v", err)
+	}
+
+	if len(item0Res.Value) == 0 {
+		t.Fatalf("Expected non-empty response, got %v", item0Res.Value)
+	}
+
+	item1Res, err = container.ReadItem(context.TODO(), pkBeta, "1", nil)
+	if err != nil {
+		t.Fatalf("Failed to read item: %v", err)
+	}
+
+	if len(item1Res.Value) == 0 {
+		t.Fatalf("Expected non-empty response, got %v", item1Res.Value)
+	}
+
+	var item0ResBody map[string]interface{}
+	err = json.Unmarshal(item0Res.Value, &item0ResBody)
+
+	if err != nil {
+		t.Fatalf("Failed to unmarshal item response: %v", err)
+	}
+
+	if item0ResBody["id"] != "1" {
+		t.Fatalf("Expected id to be 1, got %v", item0ResBody["id"])
+	}
+
+	if item0ResBody["type"] != "alpha" {
+		t.Fatalf("Expected type to be alpha, got %v", item0ResBody["type"])
+	}
+
+	if item0ResBody["value"] != "0" {
+		t.Fatalf("Expected value to be 0, got %v", item0ResBody["value"])
+	}
+
+	var item1ResBody map[string]interface{}
+	err = json.Unmarshal(item1Res.Value, &item1ResBody)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal item response: %v", err)
+	}
+
+	if item1ResBody["id"] != "1" {
+		t.Fatalf("Expected id to be 1, got %v", item1ResBody["id"])
+	}
+
+	if item1ResBody["type"] != "beta" {
+		t.Fatalf("Expected type to be beta, got %v", item1ResBody["type"])
+	}
+
+	if item1ResBody["value"] != "0" {
+		t.Fatalf("Expected value to be 0, got %v", item1ResBody["value"])
+	}
+
+	pkAll := NewPartitionKey().AppendString("1")
+	pager := container.NewQueryItemsPager("SELECT * FROM c", pkAll, nil)
```

Comment:
The Go SDK currently only supports single-partition queries. For hierarchical partition keys, I believe that means you'll have to provide all the partition keys. Any query that specifies fewer than the total number of partition keys is considered a cross-partition query and requires additional support from the SDK (which Go does not currently provide).

Reply:
@Pietrrrek I suspect this is the cause of the remaining failure. If you can update this to be a single-partition query, then I think we can get this landed!